pylint/message/message_handler_mix_in.py

```diff
@@ -257,6 +257,33 @@ def add_message(  # type: ignore # MessagesHandlerMixIn is always mixed with PyL
                 message_definition, line, node, args, confidence, col_offset
             )
 
+    def add_ignored_message(  # type: ignore # MessagesHandlerMixIn is always mixed with PyLinter
+        self: "PyLinter",
+        msgid: str,
+        line: int,
+        node: nodes.NodeNG,
+        confidence: Optional[Confidence] = UNDEFINED,
+    ) -> None:
+        """Prepares a message to be added to the ignored message storage
+
+        When earlier checks make it so add_message() is never called
+        useless-suppression will create false-positives.
+        This function avoids this by adding those message to the ignored msgs attribute
+        """
+        message_definitions = self.msgs_store.get_message_definitions(msgid)
+        for message_definition in message_definitions:
+            self.check_message_definition(message_definition, line, node)
+            self.file_state.handle_ignored_message(
```

Comment:
I think I'm going to make another PR with a refactor of `FileState.handle_ignored_message()`. There is no need to pass `node`, `args` and `confidence` here and I don't think there is a valid reason for keeping the parameters in there.
This will in return also make the arguments needed for `add_ignored_message` more intuitive.